src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -202,13 +191,28 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				this.handleDefaultOnConnect(params, profile).catch(err => onUnexpectedError(err));
 			} else {
 				profile.serverName = trim(profile.serverName);
+				this.updatePortAndProvider(profile);
 				this._connectionManagementService.addSavedPassword(profile).then(async (connectionWithPassword) => {
 					await this.handleDefaultOnConnect(params, connectionWithPassword);
 				}).catch(err => onUnexpectedError(err));
 			}
 		}
 	}
 
+	private updatePortAndProvider(profile: IConnectionProfile): void {
```

Comment:
why do we need to append the port to the server name for this fix?

Reply:
nevermind, I see you've just moved existing code snippet into a function.  I guess the CMS server runs on a different port than 1433.